consumer/src/solana.rs

```diff
@@ -645,50 +647,81 @@ impl<'a> TransferBackend for UncompressedRef<'a> {
 }
 
 #[async_trait]
-impl<'a> TransferBackend for CompressedRef<'a> {
+impl<'a> TransferBackend<compression_leafs::Model, TransferCompressedMintV1Addresses>
+    for CompressedRef<'a>
+{
     async fn transfer(
         &self,
-        collection_mint: &collection_mints::Model,
+        compression_leaf: &compression_leafs::Model,
         txn: TransferMetaplexAssetTransaction,
-    ) -> hub_core::prelude::Result<TransactionResponse<TransferAssetAddresses>> {
+    ) -> hub_core::prelude::Result<TransactionResponse<TransferCompressedMintV1Addresses>> {
         let TransferMetaplexAssetTransaction {
             recipient_address,
             owner_address,
-            collection_mint_id,
             ..
         } = txn;
         let payer = self.0.treasury_wallet_address;
         let recipient = recipient_address.parse()?;
         let owner = owner_address.parse()?;
 
-        let asset_id = todo!("wait where's the asset address");
-        let asset = self
-            .0
-            .asset_rpc_client
-            .get_asset(asset_id)
+        let asset_api = &self.0.asset_rpc();
+
+        let tree_authority_address = Pubkey::from_str(&compression_leaf.tree_authority)?;
+        let merkle_tree_address = Pubkey::from_str(&compression_leaf.merkle_tree)?;
+
+        let asset_id = compression_leaf
+            .asset_id
+            .clone()
+            .ok_or(SolanaAssetIdError::NotFound)?;
+        let asset = asset_api
+            .get_asset(&asset_id)
             .await
-            .context("Error getting asset data")?;
+            .context("fetching asset from DAA")?;
+        let asset_proof = asset_api
+            .get_asset_proof(&asset_id)
+            .await
+            .context("fetching asset proof from DAA")?;
+
+        let root: Vec<u8> = asset_proof.root.into();
+        let data_hash: Vec<u8> = asset.compression.data_hash.context("no data hash")?.into();
+        let creator_hash: Vec<u8> = asset
+            .compression
+            .creator_hash
+            .context("no creator hash")?
+            .into();
+        let leaf_id = asset.compression.leaf_id;
+        let proofs = asset_proof
+            .proof
+            .into_iter()
+            .map(|proof| Ok(AccountMeta::new_readonly(proof.try_into()?, false)))
+            .collect::<Result<Vec<AccountMeta>>>()?;
```

Comment:
We don't need to pass all the proofs but I couldn't figure out how to calculate the proof size from the info in the concurrent tree header account.

```rust
        let merkle_tree_account_data = solana_rpc.get_account_data(&merkle_tree_address)?;
        let merkle_tree_account =
            ConcurrentMerkleTreeHeader::try_from_slice(&merkle_tree_account_data)?;
        let max_buffer_size = merkle_tree_account.get_max_buffer_size();
        let max_depth = merkle_tree_account.get_max_depth();
        info!("max buffer size: {}", max_buffer_size);
        info!("max depth: {}", max_depth);
```

The js sdk has helper for it but I didn't see how I mimic it with rust.

https://github.com/solana-labs/solana-program-library/blob/17918296f847c0173fb50dc79de5c9e851de4408/account-compression/sdk/src/accounts/ConcurrentMerkleTreeAccount.ts#L141